core/objects.go

```diff
@@ -325,6 +325,9 @@ type Challenge struct {
 	// Used by dvsni challenges
 	Validation *jose.JsonWebSignature `json:"validation,omitempty"`
 
+	// used by tls-sni-01
+	N int64
```

Comment:
It would be nice to have a more descriptive name for this.  You can still have it show up as `n` in the JSON with an annotation.